### PR TITLE
Create base model with peroperty relations

### PR DESCRIPTION
**Pull request by rokytskyi**

### CODE REVIEW

**Review by SergiiBondarchuk — APPROVED** [review]

File: src/Entities/Entity.php [hunk 1]
```diff
@@ -9,53 +9,20 @@
...
-
-        return $this;
-    }
-    #endregion
-
     #region [Public methods]
 
     /**
-     * @return array
+     * @return string
```
Comment (sauron918) [inline src/Entities/Entity.php:24]: Why did you change return type to string?

```suggestion
     * @return array
```

File: src/Lib/BaseModel.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    use InteractsWithProperties;
+
+    /**
+     * BaseModel constructor.
+     * @param array $data
+     */
+    public function __construct(array $data = [])
+    {
+        if (empty($data)) {
+            return $this;
```
Comment (sauron918) [inline src/Lib/BaseModel.php:24]: Constructors don't get return values

**Review by SergeyKondratovec — APPROVED** [review]

File: src/Lib/Helpers/ArrayHelper.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * Class ArrayHelper
+ * @package SphereMall\MS\Lib\Helpers
+ */
+class ArrayHelper
+{
+    /**
+     * @param $data
+     * @param bool $recursive
+     * @return array
```
Comment (sauron918) [inline src/Lib/Helpers/ArrayHelper.php]: If we put string 'abc' or integer value we won't get an array as result. So:

```suggestion
     * @return mixed|array
```

File: src/Entities/Entity.php (hunk 1 above)
Comment (rokytskyi) [reply src/Entities/Entity.php:24]: It is for getType() method

File: src/Lib/BaseModel.php (hunk 2 above)
Comment (rokytskyi) [reply src/Lib/BaseModel.php:24]: It was in Entity constructor. I don't want to change it.
